flintrock.py

```diff
@@ -526,42 +526,98 @@ def launch_ec2(
         vpc_id=vpc_id,
         region=region)
 
+    num_instances = num_slaves + 1
     try:
         connection = boto.ec2.connect_to_region(region_name=region)
 
-        print("Launching {c} instances...".format(c=num_slaves + 1))
-
-        reservation = connection.run_instances(
-            image_id=ami,
-            min_count=(num_slaves + 1),
-            max_count=(num_slaves + 1),
-            key_name=key_name,
-            instance_type=instance_type,
-            placement=availability_zone,
-            security_group_ids=[sg.id for sg in security_groups],
-            subnet_id=subnet_id,
-            placement_group=placement_group,
-            tenancy=tenancy,
-            ebs_optimized=ebs_optimized,
-            instance_initiated_shutdown_behavior=instance_initiated_shutdown_behavior)
-
-        time.sleep(10)  # AWS metadata eventual consistency tax.
-
-        # TODO: Move this to a reusable function and add a limit on
-        #       wait time.
-        while True:
-            for instance in reservation.instances:
-                if instance.state == 'running':
-                    continue
+        if spot_price is not None:
+            print("Requesting {c} spot instances at a max price of {p}".format(c=num_instances, p=spot_price))
+
+            instance_reqs = connection.request_spot_instances(
+                price=spot_price,
+                image_id=ami,
+                count=(num_instances),
+                key_name=key_name,
+                instance_type=instance_type,
+                placement=availability_zone,
+                security_group_ids=[sg.id for sg in security_groups],
+                subnet_id=subnet_id,
+                placement_group=placement_group,
+                ebs_optimized=ebs_optimized)
+
+            spot_instance_request_ids = [sir.id for sir in instance_reqs]
+
+            print("Waiting for spot instances to be granted...")
```

Comment:
Do we need this second print here if we've already told the user we've requested spot instances? Maybe we should just add some ellipsis to the print in [L534](https://github.com/nchammas/flintrock/pull/35/files#diff-1ec778e3c6cde832d96a7c08ba31f98eR534) to suggest we are waiting and get rid of this print here.